### PR TITLE
Add fsGroup for jobs

### PR DESCRIPTION
**Pull request by d-kuro**
refs: https://github.com/cybozu-go/moco/issues/438

Add fsGroup and fsGroupChangePolicy to Pod specs.

### CODE REVIEW

File: controllers/mysqlcluster_controller.go [hunk 1]
```diff
@@ -899,6 +899,10 @@ func (r *MySQLClusterReconciler) reconcileV1StatefulSet(ctx context.Context, req
 	podSpec.InitContainers = nil
 	podSpec.WithContainers(containers...)
 	podSpec.WithInitContainers(initContainers...)
+	podSpec.WithSecurityContext(corev1ac.PodSecurityContext().
```
Comment (masa213f) [inline controllers/mysqlcluster_controller.go:902]: This logic has a breaking change.
If a MySQLCluster has a user-defined SecurityContext, moco will not reflect it.
Please merge the settings.

```yaml
apiVersion: moco.cybozu.com/v1beta2
kind: MySQLCluster
metadata:
  name: test
spec:
  podTemplate:
    spec:
      securityContext:         # user-defined securityContext
        seccompProfile:        #
          type: RuntimeDefault #
      containers:
      - name: mysqld
        image: quay.io/cybozu/mysql:8.0.28
    # Specify minimum resources so as not to overwhelm CI resources.
  volumeClaimTemplates:
  - metadata:
      name: mysql-data
    spec:
      accessModes: [ "ReadWriteOnce" ]
      resources:
        requests:
          storage: 1Gi
  logRotationSchedule: "@every 10s"
```

**Review by masa213f — CHANGES_REQUESTED** [review]
This documentation is no longer needed. Please remove it.
https://github.com/cybozu-go/moco/blob/main/docs/troubles.md

File: controllers/mysqlcluster_controller.go (hunk 1 above)
Comment (d-kuro) [reply controllers/mysqlcluster_controller.go]: Thanks, I fix and add tests.
https://github.com/cybozu-go/moco/pull/443#:~:text=Fix%20pod%20security%20contexts%20not%20being%20merged

**Comment by d-kuro** [comment]
> This documentation is no longer needed. Please remove it.
> https://github.com/cybozu-go/moco/blob/main/docs/troubles.md

removed: https://github.com/cybozu-go/moco/pull/443/commits/076418a2ccca729d8b81895b78c82b95118ceee5

**Review by masa213f — APPROVED** [review]
LGTM

**Pull request merged**